### PR TITLE
Validate against spec

### LINKED ISSUE #26: [META 27] Align ecs-logging-python
See https://github.com/elastic/ecs-logging/issues/27 for details on alignment.

- [x] Implementation
- [x] Spec validation (#31)
- [x] OSS compliance
- [x] Docs (#29)

### PR DESCRIPTION
**Pull request by basepi**
This adds validation against the upstream [spec](https://github.com/elastic/ecs-logging/blob/master/spec/spec.json).

Fixes #26

### CODE REVIEW

**Comment by basepi** [comment]
I'll fix up the datetime issues on Monday.

**Review by beniwohli — APPROVED** [review]
Looks great! Just one small detail

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                        raise ValidationError(
+                            "Value {0} for key {1} doesn't parse as an ISO datetime".format(
+                                data[k], k
+                            )
+                        )
+            if v.get("index"):
+                index = v.get("index")
+                key = json.loads(
+                    data_json.split(",")[index].split(":")[0].strip().lstrip("{")
+                )
```
[the remark this answers is not in the document]
Comment (beniwohli) [reply tests/conftest.py:64]: ah, I see we run Python 2.7 tests, so we'd need to use `OrderedDict`

[the remark this answers is not in the document]
Comment (basepi) [reply tests/conftest.py]: Didn't know about `object_pairs_hook`, thanks!

**Pull request merged**